web/controllers/project_user_controller.ex

```diff
@@ -27,4 +29,25 @@ defmodule CodeCorps.ProjectUserController do
   def handle_update(_conn, model, attributes) do
     model |> ProjectUser.update_changeset(attributes)
   end
+
+  defp schedule_email(conn, _) do
+    conn
+    |> Plug.Conn.register_before_send(&maybe_send_email/1)
+  end
+
+  defp maybe_send_email(%{assigns: %{ changeset: changeset}} = conn) do
+    case {changeset |> Ecto.Changeset.get_change(:role), changeset.data.role} do
+      {"contributor", "pending"} -> send_acceptance_email(conn.assigns.project_user)
+      _ -> nil
+    end
+
+    conn
+  end
+  defp maybe_send_email(conn), do: conn
```

Comment:
I think it needs to be done in a separate module. Not much complexity should be happening in the controller itself.